tor/core/inbox.py

```diff
@@ -157,50 +148,55 @@ def check_inbox(cfg):
     for item in reversed(list(cfg.r.inbox.unread(limit=None))):
         # Very rarely we may actually get a message from Reddit itself.
         # In this case, there will be no author attribute.
-        if item.author is None:
+        author_name = item.author.name if item.author else None
+        if author_name is None:
             send_to_modchat(
                 f'We received a message without an author -- '
                 f'*{item.subject}*:\n{item.body}', cfg
             )
             item.mark_read()
 
-        elif item.author.name == 'transcribot':
+        elif author_name == 'transcribot':
             item.mark_read()
 
-        elif item.author.name in cfg.redis.smembers('blacklist'):
+        elif cfg.redis.sismember('blacklist', author_name):
             logging.info(
-                f'Skipping inbox item from {item.author.name} who is on the '
+                f'Skipping inbox item from {author_name!r} who is on the '
```

Comment:
Not sure what `author_name!r` is -- is that some formatting that I'm unfamiliar with or a typo?

Reply:
For those following along, it's a conversion operator in the formatting syntax spec for python: https://docs.python.org/3/library/string.html#format-string-syntax